src/bin/add/args.rs

```diff
@@ -115,6 +115,10 @@ pub struct Args {
     #[structopt(long = "allow-prerelease")]
     pub allow_prerelease: bool,
 
+    /// Space-separated list of features to add.
+    #[structopt(long = "features")]
+    pub features: Option<String>,
```

Comment:
If we are following cargo, please make sure this flag does the same as https://github.com/rust-lang/cargo/blob/7dec94d387c9ed15e2399803ffc00e20b4a20cea/src/cargo/util/command_prelude.rs#L226. Multiple values can be a bit tricky in clap.

Reply:
It looks like the implementation already does approximately what you are suggesting, but it causes nonsensical behaviour. I added a straw-man test here: https://github.com/killercup/cargo-edit/pull/395/files#diff-2ef0929a94fc19f85e22db07be135054R982-R993 , but I will have to ponder for a bit before I can decide what the correct behaviour should be.

`cargo build --features thing1 thing2` will explode with:
```
error: Found argument 'thing2' which wasn't expected, or isn't valid in this context
```
I kind-of think that cargo-edit should also explode if you specify multiple positional arguments along with --features.

Reply:
Yes, approximately. But not exactly.

Check this to see how structopt types works. https://docs.rs/structopt/0.3.12/structopt/#type-magic

What you need to use is `Option<Vec<String>>` and then add `number_of_values = 1` to the attribute.

```rust
    #[structopt(long = "features", number_of_values = 1)]
    pub features: Option<Vec<String>>,
```

Reply:
I tried your suggestion, but I couldn't observe any behavioural changes, so I reverted it back to `Option<String>` for simplicity. Can you suggest any command-line input for which the two different implementations will behave differently?

Reply:
`number_of_values` is important. The reason is given [here](https://github.com/rust-lang/cargo/blob/7dec94d387c9ed15e2399803ffc00e20b4a20cea/src/cargo/util/command_prelude.rs#L228). You might not see any issues on the CLI.

Reply:
ah. I see what you're getting at: `--features nose --features mouth"`. I have added a test for this now.